README.md

```diff
@@ -47,13 +47,33 @@ You can also set `-Zmiri-start-fn` to make Miri start evaluation with the
 
 ## Running Miri on your own project('s test suite)
 
-Install Miri as a cargo subcommand with `cargo install --all-features`, and install
-a full libstd as described above.
+Install Miri as a cargo subcommand with `cargo install --all-features --path .`.
 
-Then, inside your own project, use `MIRI_SYSROOT=~/.xargo/HOST cargo +nightly
-miri` to run your project, if it is a bin project, or run
-`MIRI_SYSROOT=~/.xargo/HOST cargo +nightly miri test` to run all tests in your
-project through Miri.
+Compile your project and its dependencies against a MIR-enabled libstd as described
+above:
```

Comment:
A footgun here is that the `rust-toolchain` file in the miri repo will mean most people will build the libstd with the nightly we pinned for miri, which is likely not the one they pinned everywhere else.

With miri being a project that's intended to be run outside of its directory, but being sensitive to the toolchain, I feel now that using `rust-toolchain` was a mistake. Once we get rid of that, it should be much less likely that people trip when executing miri on their projects.  Then the instructions above could be changed to `rustup run nightly xargo/build.sh` and `MIRI_SYSROOT=~/.xargo/HOST cargo +nightly run tests/run-pass-fullmir/hashmap.rs`, and they can explain that the toolchain matters and libstd needs to be rebuilt when it changes.

Reply:
I've updated the instructions to list all the different compiler error messages that I could think of, and what the likely cause is. Should I remove `rust-toolchain` or add a note to the README for now that warns about it?

Reply:
> I've updated the instructions to list all the different compiler error messages that I could think of

Nice. :)